hawkular-alerts-engine/src/main/java/org/hawkular/alerts/engine/impl/CassDefinitionsServiceImpl.java

```diff
@@ -225,6 +129,16 @@ public void addActionDefinition(String tenantId, ActionDefinition actionDefiniti
         if (isEmpty(actionDefinition.getProperties())) {
             throw new IllegalArgumentException("Properties must be not null");
         }
+        String plugin = actionDefinition.getActionPlugin();
+        if (!getActionPlugins().contains(plugin)) {
+            throw new IllegalArgumentException("Plugin: " + plugin + " is not deployed");
+        }
+        Set<String> pluginProperties = getActionPlugin(plugin);
+        actionDefinition.getProperties().keySet().stream().forEach(property -> {
+            if (!pluginProperties.contains(property)) {
+                throw new IllegalArgumentException("Property: " + property + " is not valid on plugin: " + plugin);
+            }
```

Comment:
I'm not sure if throwing an exception is too strong, perhaps a warning log message would be sufficient, but I am OK with it.  I can't recall, can action plugins declare required properties?  If so, I don't see a check to ensure required properties are provided, only whether provided properties are valid.